base/gmp.jl

```diff
@@ -69,6 +69,96 @@ function __init__()
     end
 end
 
+
+module MPZ
+# wrapping of libgmp functions
+# "output parameters" are labeled x, y, z, and are returned when appropriate
+# constant input parameter are labeled a, b, c
+
+using Base.GMP: BigInt, Limb
+
+gmpz(op::Symbol) = (Symbol(:__gmpz_, op), :libgmp)
+
+sizeinbase(a::BigInt, b) = Int(ccall((:__gmpz_sizeinbase, :libgmp), Csize_t, (Ptr{BigInt}, Cint), &a, b))
+
+for op in (:add, :sub, :mul, :fdiv_q, :tdiv_q, :fdiv_r, :tdiv_r, :gcd, :lcm, :and, :ior, :xor)
+    @eval begin
+        $op(x::BigInt, a::BigInt, b::BigInt) = (ccall($(gmpz(op)), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b); x)
```

Comment:
@StefanKarpinski and @tkelman, could you comment on this? I kept the names as in libgmp without any `!` even for mutating functions (and for few functions I added a variant with a `!` when updating the input, like `add!(x, y) = add(x, x, y)`, equivalent to a `x += y`), which seems more predictable for me, but @thofma suggests to add `!` whenever an argument is modified.